controller/konnect/reconciler_generic.go

```diff
@@ -167,6 +167,9 @@ func (r *KonnectEntityReconciler[T, TEnt]) Reconcile(
 					if k8serrors.IsConflict(err) {
 						return ctrl.Result{Requeue: true}, nil
 					}
+					if k8serrors.IsNotFound(err) {
+						return ctrl.Result{}, nil
+					}
```

Comment:
As commented elsewhere: I believe we can refactor all these into a single function, WDYT?

IMO returning no error here when the updated entity is not found (during finalizer removal) is OK: there's nothing more to be done now, the CR is gone.

Reply:
Thanks, updated https://github.com/Kong/kong-operator/pull/2625/commits/c717e2b8d7ee23bdb9488fc256b2dbc643131177